app/app/core/security.py

```diff
@@ -8,20 +8,46 @@
 
 pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
 
+ALGORITHM = "HS512"
 
-ALGORITHM = "HS256"
 
-
-def create_access_token(subject: Union[str, Any], expires_delta: timedelta = None) -> str:
+def create_token(subject: Union[str, Any], expires_delta: timedelta = None, type_: str = 'access') -> str:
     if expires_delta:
         expire = datetime.utcnow() + expires_delta
     else:
-        expire = datetime.utcnow() + timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)
-    to_encode = {"exp": expire, "sub": str(subject)}
+        expire = datetime.utcnow() + get_timedelta_for_type(type_)
+    to_encode = {"exp": expire, "sub": str(subject), "type": type_}
     encoded_jwt = jwt.encode(to_encode, settings.SECRET_KEY, algorithm=ALGORITHM)
     return encoded_jwt
 
 
+def get_timedelta_for_type(type_: str) -> timedelta:
+    """
+    :param type_: Type of access token
+    :return: Expiry time for token, in minutes
+    """
+    if type_ == 'refresh':
+        return timedelta(minutes=settings.REFRESH_TOKEN_EXPIRES_MINUTES)
+    if type_ == 'access':
+        return timedelta(minutes=settings.ACCESS_TOKEN_EXPIRE_MINUTES)
+    return timedelta(minutes=5)
+
+
+def create_access_token(subject: Union[str, Any], expires_delta: timedelta = None) -> str:
```

Comment:
I don't feel like we need to have two separate functions, just a single `create_token` that takes in `type_` should be enough. We can then set `expires_delta` according to the given `type_`.